drv/i2c-devices/src/max5970.rs

```diff
@@ -232,8 +240,81 @@ impl Max5970 {
 
 impl Validate<ResponseCode> for Max5970 {
     fn validate(device: &I2cDevice) -> Result<bool, ResponseCode> {
-        let val = Max5970::new(device).read_reg(Register::cbuf_dly_stop)?;
-
+        let val = Max5970::new(device, 0, Ohms(0.0))
+            .read_reg(Register::cbuf_dly_stop)?;
         Ok(val == 0x19)
     }
 }
+
+impl VoltageSensor<ResponseCode> for Max5970 {
+    fn read_vout(&self) -> Result<Volts, ResponseCode> {
+        let (msb, lsb) = if self.rail == 0 {
+            (
+                self.read_reg(Register::adc_chx_mon_msb_ch1)?,
+                self.read_reg(Register::adc_chx_mon_lsb_ch1)?,
+            )
+        } else {
+            (
+                self.read_reg(Register::adc_chx_mon_msb_ch2)?,
+                self.read_reg(Register::adc_chx_mon_lsb_ch2)?,
+            )
+        };
+
+        Ok(Volts(((((msb as u16) << 2) | (lsb as u16)) as f32) / 64.0))
```

Comment:
I notice that `read_iout` checks the status register to decide on the divider, but this function locks it to 64.

Should we check the divider here, e.g. (untested)
```rust
        let mon_range = self.read_reg(Register::mon_range)?;

        let range = if self.rail == 0 {
            mon_range & 0b11
        } else {
            (mon_range >> 2) & 0b11
        };

        let full_scale_voltage = match range {
            0b00 => 16,
            0b01 => 8,
            0b10 => 4,
            0b11 => 2,
        };
        
        let scale = (1024 / full_scale_voltage) as f32;
        let value = ((msb as u16) << 2) | (lsb & 0b11 as u16);
        Ok(Volts(value / scale))
```

(I'm also explicitly masking `lsb`, to make it obvious that we're intentionally only using two bits)

Reply:
Done -- thank you!